### PR TITLE
[Doc] [KubeRay] [RayJob] Add info about submitter pod template

### PR DESCRIPTION
**Pull request by architkulkarni**
## Why are these changes needed?
Adds information about the JobSubmitterPod to the RayJob documentation.  This is an advanced configuration option that was previously only documented in comments in the sample RayJob YAML file.

## Related issue number
Resolves https://github.com/ray-project/kuberay/pull/1319#pullrequestreview-1660564132

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/cluster/kubernetes/getting-started/rayjob-quick-start.md [hunk 1]
```diff
@@ -147,3 +147,9 @@ helm uninstall kuberay-operator
 # Step 6.3: Delete the Kubernetes cluster
 kind delete cluster
 ```
+
+## Advanced Usage
+
+The Pod template for the Kubernetes Job that runs `ray job submit` can be customized by setting the `submitterPodTemplate` field in the RayJob custom resource.  See <https://raw.githubusercontent.com/ray-project/kuberay/master/ray-operator/config/samples/ray_v1alpha1_rayjob.yaml> for an example (commented out in this file).
+
+If `submitterPodTemplate` is unspecified, the pod will consist of a container named `ray-job-submitter` with image matching that of the Ray head, resource requests of 500m CPU and 200MiB memory, and limits of 1 CPU and 1GiB memory.
```
Comment (kevin85421) [inline doc/source/cluster/kubernetes/getting-started/rayjob-quick-start.md:155]:
```suggestion
If `submitterPodTemplate` is unspecified, the Pod will consist of a container named `ray-job-submitter` with image matching that of the Ray head, resource requests of 500m CPU and 200MiB memory, and limits of 1 CPU and 1GiB memory.
```

Comment (kevin85421) [reply doc/source/cluster/kubernetes/getting-started/rayjob-quick-start.md:155]: Would you mind adding a link to your benchmark result?

**Review by kevin85421 — APPROVED** [review]

File: doc/source/cluster/kubernetes/getting-started/rayjob-quick-start.md (hunk 1 above)
Comment (architkulkarni) [reply doc/source/cluster/kubernetes/getting-started/rayjob-quick-start.md]: The benchmark was kind of informal so I don't know if it makes sense to include it in the user-facing documentation. Would you be okay with merging this PR without the benchmark? If you like I can add the link to the benchmark in the KubeRay repo as a code comment.

Comment (kevin85421) [reply doc/source/cluster/kubernetes/getting-started/rayjob-quick-start.md]: Both are fine.

**Review by hongchaodeng — APPROVED** [review]
LGTM!

**Comment by architkulkarni** [comment]
GPU doctest failure unrelated. (This PR just adds markdown text)
Linkcheck unreltaed (no links were modified)
Windows tune failures unrelated

**Comment by architkulkarni** [comment]
Screenshot 2023-10-19 at 3 40 41 PM
Looks correct

**Pull request merged**